indexer/src/main/java/au/org/aodn/esindexer/service/IndexerServiceImpl.java

```diff
@@ -204,7 +216,20 @@ protected StacCollectionModel getMappedMetadataValues(String metadataValues) thr
         }
 
         // organisation vocabs
-        // TODO: the logics for mapping record's organisation vocabs are heavily customised for a manual approach, AI now or later? need dedicated service's method
+        Set<String> mappedOrganisationLabels = new HashSet<>();
+        List<String> organisationLabelsFromThemes = vocabService.extractOrganisationVocabLabelsFromThemes(stacCollectionModel.getThemes());
+        if (!organisationLabelsFromThemes.isEmpty()) {
+            mappedOrganisationLabels.addAll(organisationLabelsFromThemes);
+        } else {
+            // manual mapping with custom logics when the record doesn't have existing AODN Organisation Vocabs
+            List<VocabModel> mappedOrganisationVocabsFromContacts = vocabService.getMappedOrganisationVocabsFromContacts(stacCollectionModel.getContacts());
+            if (!mappedOrganisationVocabsFromContacts.isEmpty()) {
```

Comment:
Need empty check?

Reply:
there is a for loop this one, maybe need empty check before trying to run the for loop?

Reply:
the for loop is ok if your list is empty, it will just skip you can give it a try .